presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -505,16 +524,112 @@ public void testSchemaAuthorizationForRole()
+                     .build())
+                .build();
+
+        Session anotherUser = testSessionBuilder()
+                .setCatalog(getSession().getCatalog().get())
+                .setSchema("test_createschema_authorization_user")
+                .setIdentity(Identity.forUser("anotheruser")
+                     .withPrincipal(getSession().getIdentity().getPrincipal())
+                     .build())
+                .build();
+
+        assertUpdate(admin, "CREATE SCHEMA test_createschema_authorization_user AUTHORIZATION user");
+        assertUpdate(user, "CREATE TABLE test_createschema_authorization_user.test (x bigint)");
+
+        // another user should not be able to drop the table
+        assertQueryFails(anotherUser, "DROP TABLE test_createschema_authorization_user.test", "Access Denied: Cannot drop table test_createschema_authorization_user.test");
+        // or access the table in any way
+        assertQueryFails(anotherUser, "SELECT 1 FROM test_createschema_authorization_user.test", "Access Denied: Cannot select from table test_createschema_authorization_user.test");
+
+        assertUpdate(user, "DROP TABLE test_createschema_authorization_user.test");
+        assertUpdate(user, "DROP SCHEMA test_createschema_authorization_user");
+    }
+
+    @Test
+    public void testCreateSchemaWithAuthorizationForRole()
+    {
+        Session admin = Session.builder(getQueryRunner().getDefaultSession())
+                .setIdentity(Identity.forUser("hive")
+                        .withRole("hive", new SelectedRole(ROLE, Optional.of("admin")))
+                        .build())
+                .build();
+
+        Session user = testSessionBuilder()
+                .setCatalog(getSession().getCatalog().get())
+                .setSchema("test_createschema_authorization_role")
+                .setIdentity(Identity.forUser("user")
+                     .withPrincipal(getSession().getIdentity().getPrincipal())
+                     .build())
+                .build();
+
+        Session anotherUser = testSessionBuilder()
+                .setCatalog(getSession().getCatalog().get())
+                .setSchema("test_createschema_authorization_role")
+                .setIdentity(Identity.forUser("anotheruser")
+                     .withPrincipal(getSession().getIdentity().getPrincipal())
+                     .build())
+                .build();
+
+        assertUpdate(admin, "CREATE ROLE authorized_users");
+        assertUpdate(admin, "GRANT authorized_users TO user");
+
+        assertQueryFails(admin, "CREATE SCHEMA test_createschema_authorization_role AUTHORIZATION ROLE nonexisting_role", ".*?Role 'nonexisting_role' does not exist");
+        assertUpdate(admin, "CREATE SCHEMA test_createschema_authorization_role AUTHORIZATION ROLE authorized_users");
+        assertUpdate(user, "CREATE TABLE test_createschema_authorization_role.test (x bigint)");
```

Comment:
can you try to create another table when user has this role disabled (or enabled different role)?

Reply:
Somehow I cannot set this up in the test - even it works when I test it with an installed server.

Reply:
You need to create another session instance where user has disabled role. `SET ROLE ...` does not work in unit tests (it does not modify the session object).

Reply:
I tried that too. Lemme look at that again.

Reply:
Done.